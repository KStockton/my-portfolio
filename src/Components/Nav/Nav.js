import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
 
class Nav extends Component {
  constructor() {
    super();
    this.state = {
      isChecked: false
    };
  }

  handleChange = (event) => {
    const { checked } = event.target;

    this.setState({isChecked: checked });
  }
  
  render() {
    const { isChecked } = this.state;

    return (
      <header>
        <input 
          type="checkbox"
          id="menu" 
          checked={this.state.isChecked}
          value={this.state.isChecked} 
          onChange={this.handleChange}/>
        <label htmlFor="menu">
          <div className="bar top"></div>
          <div className="bar middle"></div>
          <div className="bar bottom"></div>
        </label>
        { (isChecked) ? 
          <ul className="Home-nav">
            <NavLink to='/' id='home'>
              <li>
                <i className="fas fa-home"></i>
                Home
              </li>
            </NavLink>
            <NavLink to='/contact' className="nav">
              <li>
                <i className="fas fa-envelope"></i>
                Contact
              </li>
            </NavLink>
          </ul> : null }
      </header>
    );
  }
}

export default Nav;