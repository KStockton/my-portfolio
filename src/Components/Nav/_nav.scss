 header {
   position: relative;

   label {
    z-index: 102;
   }
 }
 
.Home-nav {
  display: flex;
  flex-direction: row;
  padding: 18px;
  font-size: 18px;
  position: fixed;
  right: 99px;
  top: 5px;
  z-index: 102;

  a {
    text-decoration: none;
    color: black;
    border-radius: 2px;
    border: none;
    box-shadow: 2px 2px 5px rgba(0, 0, 0, 0.555);
    background-color: #ffffff;
    margin-left: 3px;

    &:hover {
      background-image: linear-gradient(to right,  rgb(97, 153, 228), rgba(0, 0, 0, 0.39));
      border: 1px solid #ffd700;
      color: white;
      cursor: pointer;
    }

    &:active {
      background-image: linear-gradient(to right,  rgb(42, 196, 4), rgba(0, 0, 0, 0.39));
    }
  }
    li {
      @include displayFlex(null, null, center, row);
      list-style-type: none;
      transition: ease-in 100ms; 
      width: 100%;
      height: 100%;
      padding: 0.5rem 1.5rem;

      .fas {
        padding-right: 8px;
        text-align: center;
      }
    }
}

header label {
  padding: 12px;
  position: fixed;
  right: 41px;
  top: 16px;
  cursor: pointer;
  border: 1px solid #b3a7a7;
  background-color: #ffffff;
  box-shadow: 1px 1px 4px rgba(0, 0, 0, 0.555);
}

 #menu {
  display: none;
} 

input[type=checkbox] + label {

  .bar {
    display: block;
    width: 36px;
    height: 4px;
    background-color: black;
    margin: 4px;
    transition: all 0.2s;
  }
}

input[type=checkbox]:checked + label {
  .top {
    transform: translateY(9px) rotate(45deg);
  }
  .middle {
    opacity: 0;
  }
  .bottom {
    transform: translateY(-7px) rotate(-45deg);

  }
}

@media only screen and (max-width: 519px){
  .Home-nav {
    flex-direction: column;
    top: -5px;

    a {
      margin: 5px;
    }
  }
}