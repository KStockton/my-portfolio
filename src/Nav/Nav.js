import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
 
class Nav extends Component {
  constructor() {
    super()
    this.state = {
      checked: false
    }
  }

  handleClick = (event) => {
    this.setState({checked: !this.state.checked})
  }

  
  render() {
    const { checked } = this.state

    return (
      <header>
      <input type="checkbox" className="toggle" id="menu" value={this.state.checked} onClick={this.handleClick}/>
        <label htmlFor="menu">
          <div className="bar top"></div>
          <div className="bar middle"></div>
          <div className="bar bottom"></div>
        </label>
        { checked ? <ul className="Home-nav">
        <NavLink to='/'>
          <li>
          <i class="fas fa-home"></i>
            Home
          </li>
        </NavLink>
         <NavLink to='/portfolio' className="nav">
            <li>
            <i class="fas fa-briefcase"></i>
              Portfolio
            </li>
         </NavLink>
          <NavLink to='/contact' className="nav">
            <li>
            <i class="fas fa-envelope"></i>
              Contact
            </li>
          </NavLink>
    </ul> : null }
    </header>
  )
}
}

export default Nav;