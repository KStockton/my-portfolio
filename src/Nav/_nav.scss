 header {
   position: relative;

   label {
    z-index: 99;
   }
 }
 
.Home-nav {
  display: flex;
  flex-direction: row;
  padding: 18px;
  font-size: 18px;
  position: fixed;
  right: 83px;
  top: 5px;
  z-index: 99;

  a {
    text-decoration: none;
    color: black;
    border-radius: 2px;
    border: none;
    box-shadow: 2px 2px 5px rgba(0, 0, 0, 0.555);
    background-color: #fff;
    margin-left: 3px;

    &:hover {
      background-image: linear-gradient(to right,  rgb(97, 153, 228), rgba(0, 0, 0, 0.39));
      border: 1px solid gold;
      color: white;
      cursor: pointer;
    }

    &:active {
      background-image: linear-gradient(to right,  rgb(42, 196, 4), rgba(0, 0, 0, 0.39));
    }
  }
    li {
      @include displayFlex(center, null, center, row);
      list-style-type: none;
      transition: ease-in 100ms; 
      width: 100%;
      height: 100%;
      padding: 0.5rem 1.5rem;
    }
}

.icon {
  padding-right: 4px;
}

header label {
  padding: 12px;
  position: fixed;
  right: 18px;
  top: 16px;
  cursor: pointer;
  border: 1px solid #b3a7a7;
  background-color: #ffffffba;}

 .toggle {
  display: none;
} 

input[type=checkbox] + label {

  .bar {
    display: block;
    width: 36px;
    height: 4px;
    background-color: black;
    margin: 4px;
    transition: all 0.2s;
  }
}

input[type=checkbox]:checked + label {
  .top {
    transform: translateY(9px) rotate(45deg);
  }
  .middle {
    opacity: 0;
  }
  .bottom {
    transform: translateY(-7px) rotate(-45deg);

  }
}