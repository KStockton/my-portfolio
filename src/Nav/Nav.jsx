import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { faHome, faSuitcase, faEnvelope} from '@fortawesome/fontawesome-free-solid';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import './Nav.scss'

 
export default class Nav extends Component {
  constructor() {
    super()
    this.state = {
      checked: false
    }
  }

  handleClick = (event) => {
    this.setState({checked: !this.state.checked})
  }
  
  render() {
    let menu
    if(this.state.checked){
      menu = <nav className="Home-nav">
      <NavLink to='/'>
      <FontAwesomeIcon icon={faHome} className="icon"/>
      Home</NavLink>
      <NavLink to='/portfolio' className="nav"> 
      <FontAwesomeIcon icon={faSuitcase} className="icon"/>
      Portfolio</NavLink>
      <NavLink to='/contact' className="nav">
      <FontAwesomeIcon icon={faEnvelope} className="icon"/>
    Contact</NavLink>
    </nav>
    }
    return (
      <header>
      {menu}
      <input type="checkbox" className="toggle" id="menu" value={this.state.checked} onClick={this.handleClick}/>
        <label for="menu">
          <div className="bar top"></div>
          <div className="bar middle"></div>
          <div className="bar bottom"></div>
        </label>
    </header>
  )
}
}

// export default Header