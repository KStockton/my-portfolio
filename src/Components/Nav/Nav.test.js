import React from 'react';
import Nav from './Nav';
import { shallow } from 'enzyme';


const mockIsChecked = jest.fn();

describe('Nav', () => {
  let wrapper, mockEvent;

  beforeEach(() => {
    mockEvent = { target: { checked: true }};
    wrapper = shallow(<Nav
      handleChange={ mockIsChecked }
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should match the snapshot if isChecked is true', () => {
    wrapper.setState({ isChecked: true });
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should have a default state', () => {
    expect(wrapper.state().isChecked).toBe(false);
  });

  it('should update isChecked when called', () => {
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state().isChecked).toBe(true);
  });

  it('should call handleChange when checkbox is selected', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleChange');
    wrapper.instance().forceUpdate();
    
    wrapper.find('#menu').simulate('change', mockEvent);
    expect(spy).toHaveBeenCalled();
  });

});