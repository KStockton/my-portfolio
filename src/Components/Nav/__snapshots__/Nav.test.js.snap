// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
<header>
  <input
    id="menu"
    onClick={[Function]}
    type="checkbox"
    value={false}
  />
  <label
    htmlFor="menu"
  >
    <div
      className="bar top"
    />
    <div
      className="bar middle"
    />
    <div
      className="bar bottom"
    />
  </label>
</header>
`;
